arch/riscv/riscv_features.c

```
@@ -1,25 +1,79 @@
+#  include <asm/hwprobe.h>
+#  include <sys/syscall.h> /* For __NR_riscv_hwprobe */
+#  include <unistd.h> /* For syscall() */
+#endif
+
 #define ISA_V_HWCAP (1 << ('v' - 'a'))
 #define ISA_ZBC_HWCAP (1 << 29)
 
-void Z_INTERNAL riscv_check_features_runtime(struct riscv_cpu_features *features) {
-#if defined(__linux__) && defined(HAVE_SYS_AUXV_H)
-    unsigned long hw_cap = getauxval(AT_HWCAP);
+int Z_INTERNAL riscv_check_features_runtime_hwprobe(struct riscv_cpu_features *features) {
```

Cppcheck (2.19.0)

[style] The function 'riscv_check_features_runtime_hwprobe' should have static linkage since it is not used outside of its translation unit.

(staticFunction)